src/entities/inputValidator/inputValidator.ts

```diff
@@ -0,0 +1,49 @@
+import { AddLiquidityInput } from '../addLiquidity';
+import { CreatePoolInput } from '../createPool/types';
+import { PoolStateInput } from '../types';
+import { InputValidatorComposableStable } from './composableStable/inputValidatorComposableStable';
+import { InputValidatorGyro } from './gyro/inputValidatorGyro';
+import { InputValidatorBase } from './types';
+import { InputValidatorWeighted } from './weighted/inputValidatorWeighted';
+
+export class InputValidator {
+    validators: Record<string, InputValidatorBase> = {};
+
+    constructor() {
+        this.validators = {
+            WEIGHTED: new InputValidatorWeighted(),
+            GYRO2: new InputValidatorGyro(),
+            GYRO3: new InputValidatorGyro(),
+            GYROE: new InputValidatorGyro(),
+            PHANTOM_STABLE: new InputValidatorComposableStable(),
+        };
+    }
```

Comment:
Since we're replicating the same structure everywhere, what do you think of adding the ability to add customValidators as well?

Reply:
It wouldn't make much sense I think, since the InputValidator is not exposed to the user

Reply:
But the idea is that devs should be able to extend functionality such as addLiquidity and removeLiquidity - this means that they would want to extend their respective validator, right?
Anyway, that's basically a helper, so I'd say it's ok to leave as is and reassess in the future 👍